crates/release_plz_core/src/copy_dir.rs

```diff
@@ -1,173 +0,0 @@
-    let to = to.as_ref().join(dir_name);
-    debug!("copying directory from {:?} to {:?}", from, to);
-    if !to.exists() {
-        trace!("creating directory {:?}", to);
-        fs_err::create_dir_all(&to)?;
-    }
-
-    copy_directory(from, to)?;
-
-    Ok(())
-}
-
-/// `to` must exist.
-#[tracing::instrument]
-#[expect(clippy::filetype_is_file)] // we want to distinguish between files and symlinks
-fn copy_directory(from: &Utf8Path, to: Utf8PathBuf) -> Result<(), anyhow::Error> {
-    let walker = ignore::WalkBuilder::new(from)
-        // Read hidden files
-        .hidden(false)
-        // Don't consider `.ignore` files.
-        .ignore(false)
-        // Ignore the global `.gitignore` as it might cause issues.
-        // For example, if it contains `.git/`, we will fail in recognizing the git directory later.
-        .git_global(false)
-        .build();
-    for entry in walker {
-        let entry = entry.context("invalid entry")?;
-        let destination =
-            destination_path(&to, &entry, from).context("failed to determine destination path")?;
-        let file_type = entry.file_type().context("unknown file type")?;
-        if file_type.is_dir() {
-            if destination == to {
-                continue;
-            }
-            trace!("creating directory {:?}", destination);
-            fs_err::create_dir(&destination)?;
-        } else if file_type.is_symlink() {
-            let entry_utf8: &Utf8Path = entry.path().try_into()?;
-            let original_link = Utf8Path::read_link_utf8(entry_utf8)
-                .with_context(|| format!("cannot read link {:?}", entry.path()))?;
-            debug!("found symlink {:?} -> {:?}", entry.path(), original_link);
-            let original_link = if original_link.is_relative() {
-                original_link
-            } else {
-                let new_relative = strip_prefix(&original_link, from)?;
-                to.join(new_relative)
-            };
-            create_symlink(&original_link, &destination).with_context(|| {
-                format!(
-                    "cannot create symlink {:?} -> {:?}",
-                    &original_link, &destination
-                )
-            })?;
-        } else if file_type.is_file() {
-            trace!("copying file {:?} to {:?}", entry.path(), &destination);
-            fs_err::copy(entry.path(), &destination).with_context(|| {
-                format!("cannot copy file {:?} to {:?}", entry.path(), &destination)
-            })?;
-        }
-    }
-    Ok(())
-}
-
-fn destination_path(
-    to: &Utf8Path,
-    entry: &ignore::DirEntry,
-    from: &Utf8Path,
-) -> anyhow::Result<Utf8PathBuf> {
-    let mut dest_path = to.to_path_buf();
-    let relative = strip_prefix(entry.path().try_into()?, from)?;
-    dest_path.push(relative);
-    Ok(dest_path)
-}
-
-#[cfg(test)]
-mod tests {
-    use crate::fs_utils::Utf8TempDir;
-
-    use super::*;
-
-    #[test]
-    fn is_dir_copied_correctly() {
```

Comment:
let's bring back this beautiful tests please 😁
You can adapt them to test your copy that uses `git clone`.

Reply:
Will do!